ompi/mca/osc/ucx/osc_ucx.h

```diff
@@ -147,9 +163,20 @@ typedef struct ompi_osc_ucx_lock {
     bool is_nocheck;
 } ompi_osc_ucx_lock_t;
 
-#define OSC_UCX_GET_EP(comm_, rank_) (ompi_comm_peer_lookup(comm_, rank_)->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_UCX])
+#define OSC_UCX_GET_EP(_module, rank_) (mca_osc_ucx_component.endpoints[_module->comm_world_ranks[rank_]])
 #define OSC_UCX_GET_DISP(module_, rank_) ((module_->disp_unit < 0) ? module_->disp_units[rank_] : module_->disp_unit)
 
+extern bool opal_common_ucx_thread_enabled;
```

Comment:
This is an opal variables and should be defined in an opal header.

Reply:
Thanks for the comment, Actually, I noticed a warning message regarding this variable. I have fixed this warning in the latest push. Please kindly take look.
By the way, I added the  definition to the opal header file, and I got many more warnings like below:

````
libopen-palmca_common_ucx_noinst_la-common_ucx_wpool.o: 0000000000000001 C opal_common_ucx_thread_enabled
      osc_ucx_component.o: 0000000000000001 C opal_common_ucx_thread_enabled
  osc_ucx_active_target.o: 0000000000000001 C opal_common_ucx_thread_enabled
           osc_ucx_comm.o: 0000000000000001 C opal_common_ucx_thread_enabled
osc_ucx_passive_target.o: 0000000000000001 C opal_common_ucx_thread_enabled
        osc_ucx_request.o: 0000000000000001 C opal_common_
````

Therefore, instead, I defined it in an opal .c file, and extern'ed it in suitable places.  This is because we do not need a copy of the variable on each file that includes the opal header, instead, I defined it in a c file and extern it on other files.

Reply:
If you declare the variable in a header file you mark it as `extern` and define it once in a .c file. The messages suggest that you have definitions of the same variable in multiple files.

Reply:
I always forget the details about this warning. Here is a thread about the warning you're seeing: https://users.open-mpi.narkive.com/fTvNIGxP/ompi-make-install-warns-about-common-symbols

Bottom line: global variables (in your case `opal_common_ucx_thread_enabled`) should always be initialized where they are defined.

Reply:
Thanks @devreal I fixed this issue. 
@janjust The branch is ready.